src/fiatconnect/saga.ts

```diff
@@ -711,6 +715,60 @@ export function* handleSelectFiatConnectQuote({
   }
 }
 
+export function* handlePostKyc({ payload }: ReturnType<typeof postKycAction>) {
+  const { quote } = payload
+  const kycSchema = quote.getKycSchema()
+
+  try {
+    if (!kycSchema) {
+      // this should never happen
+      throw new Error('Unexpected error, quote is missing kyc schema')
+    }
+    yield call(postKyc, {
+      providerInfo: quote.getProviderInfo(),
+      kycSchema,
+    })
+    // We also need to save a user's quote parameters so we can re-fetch if KYC takes a long
+    // time to process.
+    yield put(
+      cacheQuoteParams({
+        providerId: quote.getProviderId(),
+        kycSchema,
+        cachedQuoteParams: {
+          cryptoAmount: quote.getCryptoAmount(),
+          fiatAmount: quote.getFiatAmount(),
+          flow: quote.flow,
+          cryptoType: quote.getCryptoType(),
+          fiatType: quote.getFiatType(),
+        },
+      })
+    )
+    // kyc will be required, but will be pending with the provider because it was
+    // just submitted.
+    yield call(_checkFiatAccountAndNavigate, { quote, isKycRequired: true, isKycApproved: false })
+    // clear persona status
+    yield put(personaFinished())
+  } catch (error) {
+    // Error while attempting to post to kyc or selecting fiat account
+    Logger.debug(
+      TAG,
+      `handlePostKyc* Error while attempting to post kyc for provider ${quote.getProviderId()}`,
+      error
+    )
+    yield put(personaFinished())
```

Comment:
I'm pretty sure this would cause the KycLanding screen to flash in the non-spinning state unless you move it to be after the navigate and a 500 delay like the other places.

Reply:
+1, also with the call above.